pymeasure/instruments/attocube/anc300.py

```diff
@@ -273,8 +278,19 @@ def __init__(
         self.ask('echo off')
 
     def check_set_errors(self):
-        """Read after setting a value."""
-        self.read()
+        """Check for errors after having set a property and log them.
+
+        Called if :code:`check_set_errors=True` is set for that property.
+
+        :return: List of error entries.
+        """
+        try:
+            self.read()
```

Comment:
So all those methods (in the instruments changed in this commit) never return an error message from `self.read()`? That's what I thought would go into the error list to return.

Reply:
In these instruments, read already does some error checking (for example checksums or the returned value) as that is needed for getting properties.
In case of some problem, read raises an exception, but does not return an error (which would be indistinguishable from a valid response).

Yes, check_errors could convert the exceptions into a returned error, but then you break symmetry: failed reading of a property is an exception, while failed setting is just a log entry.
Therefore, I decided to continue the exception path (as these instruments were previously) and not the logging path.